### DISCUSSION TITLE
Welcome to TAPAAL Discussions!

### DISCUSSION THREAD

**Discussion by yrke** [TAPAAL/TAPAAL#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by RichardJimenezG** [comment]
Hi there @yrke, is there more detailed documentation available on how to use TAPAAL than the brief documentation from http://www.tapaal.net? Thank you!

**Reply by yrke** [reply]
Not at the moment, we are currently working on updating our documentation at https://docs.tapaal.net. 
Other than that we only have an old intro tutorial (for tapaal 2.0) at [old manual](https://github.com/TAPAAL/tapaalusermanual/tree/master/manuscript). 

If you have any questions about how to use TAPAAL, feel free to post them on this forum and we will do what we can to answer.

**Reply by RichardJimenezG** [reply]
Thank you, @yrke! We have an additional question that we couldn't seem to locate in the provided documentation.

What would be the proper way to merge multiple Petri net models as part of a single system? Is it done by creating a common shared space and transitions?

Many thanks in advance!

**Reply by yrke** [reply]
Yes, in TAPAAL you can use the shared places/transition feature in combination with components to make it easier to model larger systems as individual components. I would suggest maybe having a look a the including "Alternating bit protocol" examples where you can see both a version made with shared places/transitions and one made without. 

Hope this helps

**Comment by Lerjiu** [comment]
Dear developers,
Thank you for your time in reading my comment. I'm a new user of TAPAAL 4.0 and have noticed that this version supports the creation of colored Petri nets. There are also three command-line tools available, and I'd like to know how to use them to perform CTL checking on a colored Petri net.
I haven't found any relevant information on the official website or in the documentation. I would be very grateful if you could provide a command-line example or explain the usage.
Thank you again for your time. I wish you and your family good health and success in your work.

**Reply by Tephilla** [reply]
Hi, I am not a developer, only a fellow user of TAPAAL. If you still have not figured this out yet, here is something that might be helpful.
If you are using the GUI version of TAPAAL, Under File->Example Nets->Colored P/T Nets there are a couple of colored P/T nets that you could first experiment with. 
As far as verification is concerned, you can write your query in the Queries tab that is on the left pane. 
I did not find any CLI documentation though.

**Comment by kss682** [comment]
Hello, 
I'm new to TAPAAL and have use case with Timed Coloured Petri Nets and Heirarchichal TCPNs. 
Are these options supported by TAPAAL ?
